examples/src/main/java/ai/djl/examples/training/transferlearning/TrainAmazonReviewRanking.java

```diff
@@ -120,7 +120,7 @@ public static TrainingResult runExample(String[] args)
     private static CsvDataset getDataset(
             Arguments arguments, BertFullTokenizer tokenizer, int maxLength) {
         String amazonReview =
-                "https://alpha-djl-demos.s3.amazonaws.com/dataset/nlp/amazon_reviews_us_Digital_Software_v1_00.tsv.gz";
+                "https://s3.amazonaws.com/amazon-reviews-pds/tsv/amazon_reviews_us_Digital_Software_v1_00.tsv.gz";
```

Comment:
I revert the example back. But I still think we should use our dataset:
1. In CsvDataset, we don't store the dataset locally, each time user has to re-download the gz file
2. Replacing CvsDataset with AmazonReview has no impact for user to understand this demo

Reply:
Worst case, we could at least treat it as a third-party model zoo dataset (with the URL pointing to our first party one).

I think we could use this as a standard dataset if we pull the full original one. Then, we can do the filter on the Java end to only use the digital software category (as an option to speed up the demo).

Reply:
Amazon Reviews is a collection of datasets, all different sub datasets are already stored in each .gz file: https://s3.amazonaws.com/amazon-reviews-pds/tsv/index.txt

What we kept is just the metadata that pointing to the original files. TensorFlow is doing the same thing.

Reply:
All of the data does follow the same format. So, combining the pieces should be as easy as just concatenating the data.

I think we have two options:
- Treat each of the items as a separate `Dataset` object constructed using the same class. They would all be loaded from artifacts in the same metadata file. Using multiple sub-datasets requires users to load each individually and concatenate the datasets themselves.
- Treat all of them as a unified `Dataset` object. When constructing the object, you can specify 1+ of the sub-datasets to load.

Reply:
I prefer option 1, we can all all of them into the same metadata.json file, and use `optDatasetName()` to load each of them as I did in this PR.

The whole dataset are huge, current CSVReader doesn't support streaming load, it load the whole CVS into memory. I think it make sense to load each one separately.

If user need to combine some them, they should be able to create a CombinedDataset class which is them effort as we do to implement option 2.